tokens/test/erc-1155/ERC1155Rarible.test.js

```diff
@@ -30,6 +30,485 @@ contract("ERC1155Rarible", accounts => {
+      assert.equal(await token.balanceOf(transferTo, tokenId4), mintValue);
+      //combine minted and not yet minted, try to burnBatch
+      await token.burnBatch(transferTo, [tokenId4, tokenId5], [mintValue, mintValue], {from: transferTo})
+      assert.equal(await token.balanceOf(transferTo, tokenId4), 0);//burn all minted tokenId4, burn all lazy tokenId5
+      await expectThrow(  //no lazy tokenId5, throw
+        token.mintAndTransfer([tokenId5, tokenURI, supply, creators([minter]), [], [zeroWord]], anotherUser, 1, {from: transferTo})
+      );
+    });
+
+    it("BurnBatch three tokens, ok", async () => {
+      let minter = accounts[1];
+      let anotherUser = accounts[6];
+      let transferTo = accounts[2];
+
+      const tokenId1 = minter + "b00000000000000000000001"; //save token creator
+      const tokenId2 = minter + "b00000000000000000000002"; //save token creator
+      const tokenId3 = minter + "b00000000000000000000003"; //save token creator
+      const tokenURI = "/uri";
+      let supply = 5;
+      let burn = 2;
+      let mintValue = 5;
+      //BurnBatch and after do mint
+      await expectThrow(
+        token.burnBatch(anotherUser, [tokenId1, tokenId2, tokenId3], [burn, burn, burn], {from: anotherUser})  //token has another creator
+      );
+      await expectThrow(
+        token.burnBatch(minter, [tokenId1, tokenId2, tokenId3], [burn, burn, burn], {from: anotherUser})  //burn not from minter
+      );
+      await token.burnBatch(minter, [tokenId1, tokenId2, tokenId3], [burn, burn, burn], {from: minter})  //ok
+
+      await expectThrow(  //supply - burn < mintValue == 5, throw
+        token.mintAndTransfer([tokenId1, tokenURI, supply, creators([minter]), [], [zeroWord]], transferTo, mintValue, {from: minter})
+      );
+      await token.mintAndTransfer([tokenId2, tokenURI, supply, creators([minter]), [], [zeroWord]], transferTo, 3, {from: minter})
+      assert.equal(await token.balanceOf(transferTo, tokenId2), 3);
+      await token.mintAndTransfer([tokenId3, tokenURI, supply, creators([minter]), [], [zeroWord]], transferTo, 2, {from: minter})
+      assert.equal(await token.balanceOf(transferTo, tokenId3), 2);
+    });
+
+    it("BurnBatch from different _msgFrom, ok", async () => {
+      let minter = accounts[1];
+      let anotherUser = accounts[6];
+      let transferTo = accounts[2];
+
+      const tokenId4 = minter + "b00000000000000000000004"; //save token creator
+      const tokenURI = "/uri";
+      let supply = 10;
+      let burn = 7;
+      let mintValue = 5;
+      //mint, after do burnBatch
+      await token.mintAndTransfer([tokenId4, tokenURI, supply, creators([minter]), [], [zeroWord]], transferTo, mintValue, {from: minter})
+      assert.equal(await token.balanceOf(transferTo, tokenId4), mintValue);
+      await expectThrow(  //tokenId4 send to transferTo, try to burn from anotherUser, throw
+        token.burnBatch(anotherUser, [tokenId4], [mintValue], {from: anotherUser})
```

Comment:
we should probably try to send this transaction from transferTo to check the specific require

Reply:
This transaction run later on Line 140 and if burnAmount <= lazyAmount, result ok (minter can delete only lazy)/
Add check burnAmount > lazyAmount, throw Line 136